File: Python WorldSimulatorGame/Organizm.py
import Swiat
import Punkt

class Organizm:

    def __init__(self, sila, inicjatywa, znak, wiek, zycie, polozenie: Punkt, swiat: Swiat):
        self.sila = sila
        self.inicjatywa = inicjatywa
        self.znak = znak
        self.wiek = wiek
        self.zycie = zycie
        self.polozenie = polozenie
        self.swiat = swiat

    def getSila(self):
        return self.sila

    def setSila(self, sila):
        self.sila = sila

    def getInicjatywa(self):
        return self.inicjatywa

    def setInicjatywa(self, inicjatywa):
        self.inicjatywa = inicjatywa

    def getWiek(self):
        return self.wiek

    def setWiek(self, wiek):
        self.wiek = wiek

    def getZnak(self):
        return self.znak

    def setZnak(self, znak):
        self.znak = znak

    def getZycie(self):
        return self.zycie

    def setZycie(self, zycie):
        self.zycie = zycie

    def getPolozenie(self):
        return self.polozenie

    def setPolozenie(self, polozenie):
        self.polozenie = polozenie

    def getSwiat(self):
        return self.swiat

    def setSwiat(self, swiat):
        self.swiat = swiat

    def akcja(self):
        pass

    def kolizja(self, other, atakujacy, ruch):
        pass

    def getNazwaOrganizmu(self):
        pass

    def rysowanie(self):
        pass

    def ogolneInformacje(self):
        self.swiat.dodajInformacje(str(self.getNazwaOrganizmu()) + " z pola: " + self.polozenie.printPunkt() + " ma sily: " + str(self.getSila()) + " i lat: " + str(self.getWiek()))
        #print(str(self.getNazwaOrganizmu()) + " z pola: " + self.polozenie.printPunkt() + " ma sily: " + str(self.getSila()) + " i lat: " + str(self.getWiek()) + " ", end='')

    def zabojczeInformacjeDlaZwierzat(self, other):
        self.swiat.dodajInformacje(" " + str(other.getNazwaOrganizmu()) + " zabil -> " + str(self.getNazwaOrganizmu()))
        #print(" " + str(other.getNazwaOrganizmu()) + " zabil -> " + str(self.getNazwaOrganizmu()), end='')

    def zabojczeInformacjeDlaRoslin(self, org1, org2):
        #print(" " + str(org1.getNazwaOrganizmu()) + " zjadl -> " + str(org2.getNazwaOrganizmu()), end='')
        self.swiat.dodajInformacje(" " + str(org1.getNazwaOrganizmu()) + " zjadl -> " + str(org2.getNazwaOrganizmu()))

    def informacjeONowychOrganizmach(self, nazwa, miejsceUrodzenia):
        self.swiat.dodajInformacje(" Rozmnaza sie! Powstaje nowy " + str(nazwa) + " na polu: " + miejsceUrodzenia.printPunkt())
        #print(" Rozmnaza sie! Powstaje nowy " + str(nazwa) + " na polu: " + miejsceUrodzenia.printPunkt(), end='')

File: C++ WorldSimulatorGame/POproj1/Organizm.cpp
#include "Organizm.h"

Organizm::Organizm( int sila, int inicjatywa, char znak, int wiek, bool zycie, Organizmy organizm, Punkt polozenie, Swiat* swiat) {
	this->sila = sila;
	this->inicjatywa = inicjatywa;
	this->znak = znak;
	this->wiek = wiek;
	this->zycie = zycie;
	this->organizm = organizm;
	this->polozenie = polozenie;
	this->swiat = swiat;
}

int Organizm::getSila() {
	return sila;
}
int Organizm::getInicjatywa() {
	return inicjatywa;
}
int Organizm::getWiek() {
	return wiek;
}
char Organizm::getZnak() {
	return znak;
}
bool Organizm::getZycie() {
	return zycie;
}
Organizm::Organizmy Organizm::getRodzajOrganizmu() {
	return organizm;
}

Punkt Organizm::getPolozenie() {
	return polozenie;
}
Swiat* Organizm::getSwiat() {
	return swiat;
}

void Organizm::setSila(int sila) {
	this->sila = sila;
}
void Organizm::setInicjatywa(int inicjatywa) {
	this->inicjatywa = inicjatywa;
}
void Organizm::setWiek(int wiek) {
	this->wiek = wiek;
}
void Organizm::setZnak(char znak) {
	this->znak = znak;
}
void Organizm::setZycie(bool zycie) {
	this->zycie = zycie;
}
void Organizm::setRodzajOrganizmu(Organizmy organizm) {
	this->organizm = organizm;
}
void Organizm::setPolozenie(int x, int y) {
	polozenie.setX(x);
	polozenie.setY(y);
}
void Organizm::setSwiat(Swiat* swiat) {
	this->swiat = swiat;
}
void Organizm::ogolneInformacje() {
	swiat->info.dodajInfocjmacje(this->getNazwaOrganizmu() + " z pola: " + to_string(this->getPolozenie().getX()) + " " + to_string(this->getPolozenie().getY()) + " Ma sily: " + to_string(this->getSila()));
}
void Organizm::zabojczeInformacjeDlaZwierzat(Organizm* other) {
	swiat->info.dodajInfocjmacje(" " + other->getNazwaOrganizmu() + " z sila: " + to_string(other->getSila()) + " zabil -> " + this->getNazwaOrganizmu() + " z sila: " + to_string(this->getSila()));
}
void Organizm::zabojczeInformacjeDlaRoslin(Organizm* other) {
	swiat->info.dodajInfocjmacje(" " + other->getNazwaOrganizmu() + " z sila: " + to_string(other->getSila()) + " zjadl -> " + this->getNazwaOrganizmu() + " z sila: " + to_string(this->getSila()));
}
Organizm::~Organizm(){

}